inc/dbmysql.class.php

```diff
@@ -1508,13 +1522,22 @@ public function commit() {
    }
 
    /**
-    * Rollbacks a transaction
+    * Rollbacks a transaction completely or to a specified savepoint
     *
     * @return boolean
     */
-   public function rollBack() {
-      $this->in_transaction = false;
-      return $this->dbh->rollback();
+   public function rollBack($savepoint = null) {
+      if (!$savepoint) {
+         $this->in_transaction = false;
+         $this->dbh->rollback();
+      } else {
+         $this->rollbackTo($savepoint);
```

Comment:
PHP documentation is not clear, but there is a second argument `$name` to the rollback method. Maybe it correspond to the name of the savepoint.
```suggestion
         $this->dbh->rollback(0, $savepoint);
```

Reply:
It does not. I tried that first and it did nothing. I looked at the source code for the mysqli extension and I didn't see any "ROLLBACK TO" queries anywhere. The name parameter get translated to a "ROLLBACK $name" which isn't clear from the mysql docs as to its purpose. Using named rollback didn't even through an error if naming a save point that doesn't exist like the ROLLBACK TO query does.

It is possible that it is supposed to be the same, but doesn't work properly for me.